src/pudl/analysis/ml_tools.py

```diff
@@ -0,0 +1,181 @@
+
+    return {experiment_name: model_config}
+
+
+def _flatten_model_config(model_config: dict) -> dict:
+    """Take nested dictionary defining model config and flatten for logging purposes.
+
+    This is essentially a translation layer between Dagster configuration and mlflow,
+    which does not support displaying nested parameters in the UI.
+
+    Examples:
+        >>> _flatten_model_config(
+        ...     {
+        ...         'ferc_to_ferc': {
+        ...             'link_ids_cross_year': {
+        ...                 'compute_distance_matrix': {
+        ...                     'distance_threshold': .5,
+        ...                      'metric': 'euclidean',
+        ...                 },
+        ...                 'match_orphaned_records': {'distance_threshold': 0.5},
+        ...             }
+        ...         }
+        ...     }
+        ... ) == {
+        ...     'ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.distance_threshold': 0.5,
+        ...     'ferc_to_ferc.link_ids_cross_year.compute_distance_matrix.metric': 'euclidean',
+        ...     'ferc_to_ferc.link_ids_cross_year.match_orphaned_records.distance_threshold': 0.5
+        ... }
+        True
+    """
+
+    def _flatten_level(config_level: dict, param_name: str):
+        flattened_dict = {}
+        for key, val in config_level.items():
+            flattened_param = f"{param_name}.{key}"
+            if isinstance(val, dict):
+                flattened_dict |= _flatten_level(val, param_name=flattened_param)
+            else:
+                flattened_dict[flattened_param[1:]] = val
+        return flattened_dict
+
+    return _flatten_level(model_config, "")
+
+
+class ExperimentTrackerConfig(Config):
+    """Dagster config to setup experiment tracking with mlflow."""
+
+    tracking_uri: str = f"sqlite:///{PudlPaths().output_dir}/experiments.sqlite"
+    tracking_enabled: bool = True
+    experiment_name: str
+    log_yaml: bool
+    run_context: str
+    #: Location to store artifacts. Artifact storage not currently used.
+    artifact_location: str = str(PudlPaths().output_dir)
+
+
+class ExperimentTracker(BaseModel):
+    """Class to manage tracking a machine learning model using MLflow.
+
+    The following command will launch the mlflow UI to view model results:
+    `mlflow ui --backend-store-uri {tracking_uri}`. From here, you can compare metrics
+    from multiple runs, and track performance.
+
+    This class is designed to be created using the `op` :func:`create_experiment_tracker`.
+    This allows the `ExperimentTracker` to be passed around within a Dagster `graph`,
+    and be used for mlflow logging in any of the `op`s that make up the `graph`. This
+    is useful because Dagster executes `op`s in separate processes, while mlflow does
+    not maintain state between processes. This design also allows configuration of
+    the ExperimentTracker to be set from the Dagster UI.
+
```

Comment:
How can you change the configuration from the Dagster UI? Maybe I've just never figured out how to change any of the Dagster config stuff from the UI. I think eventually it would be nice if there was an Experiment Tracking section in the PUDL Development docs but maybe that's for a later PR.

Reply:
Yeah I'm still figuring out dagster config, initially you had to create the structure yourself and it was pretty confusing to actually get right, but I've made some changes, so now if you click `Open Launchpad` in the UI, the config will pop up for each model, and you can easily edit values right there.